app/controllers/groups_controller.rb

```diff
@@ -29,11 +32,32 @@ def update
     end
   end
 
+  def new
+    @parent_groups = Group.all
+                          .includes(:children)
+                          .reject { |g| g.children.empty? }
+                          .sort_by { |g| g.children.length }
+                          .reverse
+  end
+
+  def create
+    group = Group.create(create_params)
+    leader = User.find(params[:selected_user_id])
+    membership = CreateMembership.call(group, leader)
+    membership.posts.where(post_type_id: PostType::DEFAULT_POST_ID).destroy_all
+    CreatePost.call(group,membership, PostType::LEADER_POST_ID)
+    redirect_to group_path(group)
```

Comment:
Just as a reminder from yesterday, this logic could be encapsulated. This one should take care of:
- (pre-validation - if necessary against any condition)
- create group
- create leader membership
- notify the leader maybe?

Reply:
Thanks, I will move these to a CreateGroup service object!